nvflare/app_common/aggregators/dxo_aggregator.py

```diff
@@ -152,6 +166,17 @@ def aggregate(self, fl_ctx: FLContext) -> DXO:
             DXO: the weighted mean of accepted DXOs from contributors
         """
 
+        if len(self.processed_algorithms) > 0:
+            processed_algorithms = np.unique(self.processed_algorithms)
+            if len(processed_algorithms) != 1:
+                raise ValueError(
+                    f"Only support aggregation of data processed with "
+                    f"the same type of algorithm but got mixed algorithms: {processed_algorithms}"
+                )
+            processed_algorithm = str(processed_algorithms[0])
+        else:
+            processed_algorithm = None
+
```

Comment:
since we only support 1 processed_algorithm,
maybe we can move the check in "accept" method.

```
if processed_algorithm is not None:
    if self.processed_algorithm is None:
        self.processed_algorithm = processed_algorithm
    elif self.processed_algorithm != processed_algorithm:
        self.log_error(XXX)
        return False
```

Reply:
Good point. Changed.